ledger/src/blockstore.rs

```diff
@@ -466,14 +454,14 @@ impl Blockstore {
 
     pub fn open_with_signal(
         ledger_path: &Path,
-        recovery_mode: Option<BlockstoreRecoveryMode>,
-        enforce_ulimit_nofile: bool,
+        options: BlockstoreOptions,
     ) -> Result<BlockstoreSignals> {
         let mut blockstore = Self::open_with_access_type(
             ledger_path,
-            AccessType::PrimaryOnly,
-            recovery_mode,
-            enforce_ulimit_nofile,
+            BlockstoreOptions {
+                access_type: AccessType::PrimaryOnly,
+                ..options
+            },
```

Comment:
Even though the previous code only used `AccessType::PrimaryOnly`, I feel like we should just pass the `options` struct as is to `open_with_access_type()`.

The caller will get `PrimaryOnly` if they `BlockstoreOptions::default()`; otherwise, if they specify values manually, they presumably want specific behavior and we should give them what they want.

Reply:
I think that makes sense.  `open_with_signal` itself should have no business with the access type.  Will do!